src/seer/automation/autofix/agent_context.py

```
@@ -229,23 +236,34 @@ def _get_commit_file_diffs(self) -> tuple[list[str], list[str]]:
     def _get_data(self):
         documents, nodes = self._load_data_from_github()
 
-        storage_path = f"models/autofix_storage_context/"
+        logger.debug(f"Loading index from storage context")
 
-        service_context = ServiceContext.from_defaults(embed_model=self.embed_model)
-        storage_context = StorageContext.from_defaults(
-            vector_store=MemoryVectorStore(), persist_dir=storage_path
-        )
-        index_structs = storage_context.index_store.index_structs()
-        if len(index_structs) == 0:
-            raise Exception("No index structures found in storage context")
-        index_struct: IndexDict = index_structs[0]  # type: ignore
-
-        index = VectorStoreIndex(
-            index_struct=index_struct,
-            service_context=service_context,
-            storage_context=storage_context,
-            show_progress=True,
-        )
+        storage_path = os.path.join("./", "models/autofix_storage_context/")
+
+        with sentry_sdk.start_span(
+            op="seer.automation.autofix.index_loading",
+            description="Loading the vector store index from local filesystem",
+        ) as span:
+            service_context = ServiceContext.from_defaults(embed_model=self.embed_model)
+            memory_vector_store = MemoryVectorStore().from_persist_dir(storage_path)
+            storage_context = StorageContext.from_defaults(
+                vector_store=memory_vector_store,
+                persist_dir=storage_path,
+            )
+            index_structs = storage_context.index_store.index_structs()
+
+            if len(index_structs) == 0:
+                raise Exception("No index structures found in storage context")
```

do we want to raise an exception here or do an alternate workflow

---

hmm something has gone very wrong if there are no index structs here so I think exception is the right move